esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2872,32 +2790,30 @@ bool CFileSprayEx::onFileList(IEspContext &context, IEspFileListRequest &req, IE
     return true;
 }
 
-void CFileSprayEx::queryDropZoneInfo(const char* dropZone, StringBuffer& path, EnvMachineOS& os, IpAddress& ip)
+bool CFileSprayEx::checkDropZoneIPAndPath(double clientVersion, const char* dropZoneName, const char* netAddr, const char* path)
 {
-    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
-    factory->validateCache();
-
-    Owned<IConstEnvironment> env = factory->openEnvironment();
-    if (!env)
-        throw MakeStringException(ECLWATCH_CANNOT_GET_ENV_INFO,"Cannot get environment information.");
+    if (isEmptyString(netAddr) || isEmptyString(path))
+        throw MakeStringException(ECLWATCH_INVALID_INPUT, "NetworkAddress or Path not defined.");
 
-    Owned<IPropertyTree> root = &env->getPTree();
-    VStringBuffer xpath("Software/DropZone[@name='%s']", dropZone);
-    IPropertyTree *pt = root->queryPropTree(xpath.str());
-    if (!pt)
-        throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone %s not found.", dropZone);
-
-    const char* computer = pt->queryProp("@computer");
-    IConstMachineInfo* machine = env->getMachine(computer);
-    if (!machine)
-        throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone %s: machine %s not found.", dropZone, computer);
-
-    SCMStringBuffer netAddr;
-    machine->getNetAddress(netAddr);
-    ip.ipset(netAddr.str());
+    Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory();
+    Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
+    Owned<IConstDropZoneInfoIterator> dropZoneItr = constEnv->getDropZoneIteratorByAddress(netAddr);
+    ForEach(*dropZoneItr)
+    {
+        SCMStringBuffer directory, name;
+        IConstDropZoneInfo& dropZoneInfo = dropZoneItr->query();
+        dropZoneInfo.getDirectory(directory);
+        if (directory.length() && (strnicmp(path, directory.str(), directory.length()) == 0))
+        {
+            if (isEmptyString(dropZoneName))
+                return true;
 
-    os = machine->getOS();
-    pt->getProp("@directory", path);
+            dropZoneInfo.getName(name);
+            if (strieq(name.str(), dropZoneName))
```

Comment:
Is this right if dropZoneName is blank?
In previous commit there was conditional code if supplied dropzone name was blank.